app/queries/homepage/articles_query.rb

```diff
@@ -69,10 +69,11 @@ def filter
       # would put all the previous filters in `OR` mode with tags, but what we need
       # is to only consider tags as a `OR` sub-condition
       if tags.present?
-        # `~` is the regexp operator and the `\M` modifier signifies the "end of word"
+        # `~` is the regexp operator, the `\m` modifier signifies the "beginning of word",
+        # and the `\M` modifier signifies the "end of word".
         # see https://www.postgresql.org/docs/11/functions-matching.html#FUNCTIONS-POSIX-REGEXP
         # for additional info
-        conditions = tags.map { |tag| relation.sanitize_sql_array(["cached_tag_list ~ ?", "#{tag}\\M"]) }
+        conditions = tags.map { |tag| relation.sanitize_sql_array(["cached_tag_list ~ ?", "\\m#{tag}\\M"]) }
```

Comment:
Should we keep the regexp-boundary notation consistent? In `Article.cached_tagged_with` we use `[[:<:]]` and `[[:>:]]`.

https://github.com/forem/forem/blob/d50b22014887c8857121c435ba729dcc6e39d6f9/app/models/article.rb#L207

Reply:
Great idea! I knew I had seen `[[:<:]]` somewhere in the code :-| 

I'll send an "adaptive" PR soon and tag you!